test/integration/testRowMode.js

```diff
@@ -0,0 +1,188 @@
+/*
+ * Copyright (c) 2015-2019 Snowflake Computing Inc. All rights reserved.
+ */
+const assert = require('assert');
+const testUtil = require('./testUtil');
+const RowMode = require('./../../lib/constants/row_mode');
+
+describe('Test row mode', function () {
+  this.timeout(3000)
+  let connection;
+  const sql = `select *
+               from (select \'a\' as key, 1 as foo, 3 as name) as table1
+                        join (select \'a\' as key, 2 as foo, 3 as name2) as table2 on table1.key = table2.key
+                        join (select \'a\' as key, 3 as foo) as table3 on table1.key = table3.key`
+
+  const expectedArray = ['a', 1, 3, 'a', 2, 3, 'a', 3]
+  const expectedObject = {KEY_0: 'a', FOO_1: 1, NAME: 3, KEY_3: 'a', FOO_4: 2, NAME2: 3, KEY_6: 'a', FOO_7: 3}
+
+  const testCases = [
+    {
+      name: 'test return row as array if not set in statement',
+      connectionRowMode: RowMode.ARRAY,
+      expected: expectedArray
+    },
+    {
+      name: 'test return row as array if statement parameter array',
+      rowMode: RowMode.ARRAY,
+      expected: expectedArray
+    },
+    {
+      name: 'test return row as array if statement parameter array',
+      connectionRowMode: RowMode.ARRAY,
+      rowMode: RowMode.ARRAY,
+      expected: expectedArray
+    },
+    {
+      name: 'test return row as object if row mode is not set',
+      expected: expectedObject
+    },
+    {
+      name: 'test return row as object with suffixes for duplicated columns if statement rowMode object',
+      rowMode: RowMode.OBJECT,
+      expected: expectedObject
+    },
+    {
+      name: 'test return row as object with suffixes for duplicated columns if connection rowMode object',
+      connectionRowMode: RowMode.OBJECT,
+      expected: expectedObject
+    },
+    {
+      name: 'test return row as object if connection and statement rowMode object',
+      connectionRowMode: RowMode.OBJECT,
+      rowMode: RowMode.OBJECT,
+      expected: expectedObject
+    }
+  ]
+
+  const createItCallbackStreamResult = function (testCase) {
```

Comment:
let's use explicit paramters instead of `testCase` object for those `createIt...` methods

Reply:
fixed